core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -19,162 +19,256 @@
+      $modal.open({
+        templateUrl: 'modals/editorViewSuggestion',
+        backdrop: true,
+        size: 'lg',
+        resolve: {
+          suggestionIsOpen: function() {
+            return _isSuggestionOpen();
+          },
+          suggestionIsValid: function() {
+            return _isSuggestionValid();
+          },
+          unsavedChangesExist: function() {
+            return _hasUnsavedChanges();
+          },
+          suggestionStatus: function() {
+            return $scope.activeThread.status;
+          },
+          description: function() {
+            return $scope.activeThread.suggestion.description;
+          },
+          oldContent: function() {
+            var state = explorationData.data.states[
+              $scope.activeThread.suggestion.state_name];
+            return state !== undefined ? state.content[0].value : null;
+          },
+          newContent: function() {
+            return $scope.activeThread.suggestion.state_content.value;
+          }
+        },
+        controller: [
+          '$scope', '$modalInstance', 'suggestionIsOpen', 'suggestionIsValid',
+          'unsavedChangesExist', 'suggestionStatus', 'description',
+          'oldContent', 'newContent', 'editabilityService',
+          function(
+            $scope, $modalInstance, suggestionIsOpen, suggestionIsValid,
+            unsavedChangesExist, suggestionStatus, description,
+            oldContent, newContent, editabilityService) {
+            var SUGGESTION_ACCEPTED_MSG = 'This suggestion has already been ' +
+              'accepted.';
+            var SUGGESTION_REJECTED_MSG = 'This suggestion has already been ' +
+              'rejected.';
+            var SUGGESTION_INVALID_MSG = 'This suggestion was made ' +
+              'for a state that no longer exists. It cannot be  accepted.';
+            var UNSAVED_CHANGES_MSG = 'You have unsaved changes to ' +
+              'this exploration. Please save/discard your unsaved changes if ' +
+              'you wish to accept.';
+            $scope.isOpen = suggestionIsOpen;
+            $scope.canEdit = editabilityService.isEditable();
+            $scope.canReject = $scope.canEdit && $scope.isOpen;
+            $scope.canAccept = $scope.canEdit && $scope.isOpen &&
+              suggestionIsValid && !unsavedChangesExist;
+
+            if (!$scope.canEdit) {
+              $scope.errorMessage = '';
+            } else if (!$scope.isOpen) {
+              $scope.errorMessage = suggestionStatus === 'fixed' ?
+                SUGGESTION_ACCEPTED_MSG : SUGGESTION_REJECTED_MSG;
+            } else if (!suggestionIsValid) {
+              $scope.errorMessage = SUGGESTION_INVALID_MSG;
+            } else if (unsavedChangesExist) {
+              $scope.errorMessage = UNSAVED_CHANGES_MSG;
+            } else {
+              $scope.errorMessage = '';
+            }
+
+            $scope.oldContent = (
+              oldContent || '<span style="color:red">Oops! This state no ' +
```

Comment:
Can you do this in the HTML template? It's a bit ugly (and can be a security risk) to manipulate HTML in the JS.